runtime/src/bank/check_transactions.rs

```diff
@@ -345,4 +408,273 @@ mod tests {
             .check_and_load_message_nonce_account(&message, &bank.next_durable_nonce())
             .is_none());
     }
+
+    #[test]
+    fn test_get_durable_nonce() {
```

Comment:
afaict, there are no tests for this function in `sdk`. I did my best to cover the branches in logic.